### PR TITLE
Adding support for creating and starting jobs on PR opening/reopening

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Also adds partial support for closing and syncing.

### CODE REVIEW

File: plugins/github.js [hunk 1]
```diff
@@ -1,9 +1,86 @@
...
+function pullRequestClosed(options, request, reply) {
+    const eventId = options.eventId;
+    const jobId = options.jobId;
+    const name = options.name;
+
+    // @TODO stop running build
+    Job.update({
+        id: jobId,
+        data: {
+            state: 'DISABLED'
```
Comment (d2lam) [inline plugins/github.js]: is closed the same as merged? If it's merged should we just delete the job?

File: plugins/github.js [hunk 2]
```diff
@@ -1,9 +1,86 @@
...
+        if (waterfallError) {
+            return reply(boom.wrap(waterfallError));
+        }
+
+        return reply().code(201);
+    });
+}
+
+/**
+ * Start any running builds and disable the job for an opened pull-request
```
Comment (d2lam) [inline plugins/github.js]: I'm confused by this comment

Comment (tkyi) [inline plugins/github.js]: `Stop` any running builds?

**Comment by tkyi** [comment]
👍 LGTM

File: plugins/github.js (hunk 1 above)
Comment (stjohnjohnson) [inline plugins/github.js]: I don't think we should delete the job, we want to keep the history around.  I'd be interested in a "MERGED" state.

File: plugins/github.js [hunk 3]
```diff
@@ -1,9 +1,86 @@
...
+        data: {
+            state: 'DISABLED'
+        }
+    }, (updateError) => {
+        if (updateError) {
+            return reply(boom.wrap(updateError));
+        }
+        request.log(['webhook-github', eventId, jobId], `${name} disabled`);
+
+        return reply().code(201);
```
Comment (d2lam) [inline plugins/github.js]: (200)?

Comment (d2lam) [inline plugins/github.js]: since it's an update

File: plugins/github.js [hunk 4]
```diff
@@ -26,22 +103,35 @@ function pullRequestEvent(request, reply) {
     request.log(['webhook-github', eventId], `PR #${prNumber} ${action} for ${scmUrl}`);
 
     // Possible actions
-    // "opened", "closed", "reopened", "synchronize"
-    // @TODO ignore events from "assigned", "unassigned", "labeled", "unlabeled", "edited"
+    // "opened", "closed", "reopened", "synchronize",
+    // "assigned", "unassigned", "labeled", "unlabeled", "edited"
```
Comment (d2lam) [inline plugins/github.js]: Pipeline.sync would create the 'main' job. Do we want to do that here?

**Comment by d2lam** [comment]
LGTM 👍  Waiting for the build to finish

**Pull request merged**